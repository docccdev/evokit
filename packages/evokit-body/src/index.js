import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
