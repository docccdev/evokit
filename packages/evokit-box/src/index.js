// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
