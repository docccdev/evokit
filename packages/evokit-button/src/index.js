import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
