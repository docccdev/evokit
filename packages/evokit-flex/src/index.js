import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'display',
    'content',
    'items',
    'direction',
    'wrap',
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'order',
]);

Flex.Item = FlexItem;
