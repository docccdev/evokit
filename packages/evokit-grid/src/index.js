import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
