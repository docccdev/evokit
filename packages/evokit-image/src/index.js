import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
