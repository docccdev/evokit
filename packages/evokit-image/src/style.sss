@import './mixins.sss'

@custom-media --ek-image-media-small only screen and (min-width: 480px)
@custom-media --ek-image-media-medium only screen and (min-width: 768px)
@custom-media --ek-image-media-large only screen and (min-width: 960px)
@custom-media --ek-image-media-wide only screen and (min-width: 1200px)
@custom-media --ek-image-media-huge only screen and (min-width: 1400px)

:root
    // prop 'image-round'
    --ek-image-round-xxs: 2px
    --ek-image-round-xs: 4px
    --ek-image-round-s: 6px
    --ek-image-round-m: 8px
    --ek-image-round-l: 10px
    --ek-image-round-xl: 12px
    --ek-image-round-xxl: 14px

.image
    display: inline
    outline: none
    border: 0
    max-width: 100%
    max-height: 100%

    &[alt]
        word-wrap: break-word
        word-break: break-word
        hyphens: auto

/**
 * Display:
 *
 *  - image_display_inline
 *  - image_display_none
 *
 **/

@each $value in (inline, none)
    @mixin media .image_display_$(value)
        display: $value

/**
 * Horizontal align:
 *
 *  - image_align_*
 *
 **/

@each $value in (left, center, right)
    @mixin media .image_align_$(value)
        @if $value == center
            float: none
            display: block
            margin-left: auto
            margin-right: auto
        @else
            float: $value

/**
 * Vertical align:
 *
 *  - image_valign_*
 *
 **/

@each $value in (baseline, bottom, middle, sub, super, text-bottom, text-top, top)
    @mixin media .image_valign_$(value)
        vertical-align: $value

/**
 * Mirror:
 *
 *  - image_mirror_*
 *
 **/

@mixin media .image_mirror_x
    transform: scaleX(-1)

@mixin media .image_mirror_y
    transform: scaleY(-1)

@mixin media .image_mirror_xy
    transform: scale(-1, -1)

/**
 * Fix:
 *
 *  - image_fit_*
 *
 **/

@each $value in (none, fill, contain, cover, scale-down)
    @mixin media .image_fit_$(value)
        object-fit: $value

/**
 * Width:
 *
 *  - image_width_*
 *
 **/

@each $to in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @each $from in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
        @if $from < $to or ($from == 1 and $to == 1)
            @mixin media .image_width_$(from)-$(to)
                width: calc($from / $to * 100%)

@mixin media .image_width_auto
    width: auto

/**
 * Round:
 *
 *  - image_round_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, full)
    @mixin media .image_round_$(value)
        @if $value == none
            border-radius: 0
        @else if $value == full
            border-radius: 50%
        @else
            border-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-top_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-top-right-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-top-right-radius: 50%
        @else
            border-top-left-radius: var(--ek-image-round-$(value))
            border-top-right-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-right_$(value)
        @if $value == none
            border-top-right-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-image-round-$(value))
            border-bottom-right-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-bottom_$(value)
        @if $value == none
            border-bottom-left-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-image-round-$(value))
            border-bottom-right-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-left_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-bottom-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-bottom-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-image-round-$(value))
            border-bottom-left-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-top-left_$(value)
        @if $value == none
            border-top-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-top-right_$(value)
        @if $value == none
            border-top-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-bottom-right_$(value)
        @if $value == none
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-right-radius: 50%
        @else
            border-bottom-right-radius: var(--ek-image-round-$(value))

    @mixin media .image_round-bottom-left_$(value)
        @if $value == none
            border-bottom-left-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-image-round-$(value))

/**
 * Alt:
 *
 *  - image_alt_visible
 *  - image_alt_hidden
 *
 **/

@each $value in (visible, hidden)
    @mixin media .image_alt_$(value)
        &[alt]
            @if $value == visible
                font-size: inherit
            @else if $value == hidden
                font-size: 0
