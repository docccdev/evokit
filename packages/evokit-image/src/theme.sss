@import './mixins.sss'

:root
    --ek-image-border-TEMPLATE_NAME: var(--border)
    --ek-image-background-TEMPLATE_NAME: var(--background)

@mixin media .image_background_TEMPLATE_NAME
    background: var(--ek-image-background-TEMPLATE_NAME)

@mixin media .image_border_TEMPLATE_NAME
    border: 1px solid var(--ek-image-border-TEMPLATE_NAME)

@mixin media .image_border-top_TEMPLATE_NAME
    border-top: 1px solid var(--ek-image-border-TEMPLATE_NAME)

@mixin media .image_border-right_TEMPLATE_NAME
    border-right: 1px solid var(--ek-image-border-TEMPLATE_NAME)

@mixin media .image_border-bottom_TEMPLATE_NAME
    border-bottom: 1px solid var(--ek-image-border-TEMPLATE_NAME)

@mixin media .image_border-left_TEMPLATE_NAME
    border-left: 1px solid var(--ek-image-border-TEMPLATE_NAME)

@mixin media .image_border-tb_TEMPLATE_NAME
    border-top: 1px solid var(--ek-image-border-TEMPLATE_NAME)
    border-bottom: 1px solid var(--ek-image-border-TEMPLATE_NAME)

@mixin media .image_border-lr_TEMPLATE_NAME
    border-left: 1px solid var(--ek-image-border-TEMPLATE_NAME)
    border-right: 1px solid var(--ek-image-border-TEMPLATE_NAME)
