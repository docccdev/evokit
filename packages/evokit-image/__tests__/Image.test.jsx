import React from 'react';
import { shallow } from 'enzyme';
import { Image } from '../src';

describe('<Image />', () => {
    it('render', () => {
        const wrapper = shallow(<Image src='test.jpg' alt='test alt' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<img src="test.jpg" alt="test alt" class="ek-image"/>');
    });

    it('props className', () => {
        const wrapper = shallow(<Image className='ololo' src='' alt='' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<img src="" alt="" class="ek-image ololo"/>');
    });

    it('props image-tag', () => {
        const wrapper = shallow(<Image image-tag='span' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<span class="ek-image"></span>');
    });
});
