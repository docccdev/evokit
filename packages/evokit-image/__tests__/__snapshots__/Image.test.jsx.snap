// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> props className 1`] = `
<img
  alt=""
  className="ek-image ololo"
  src=""
/>
`;

exports[`<Image /> props image-tag 1`] = `
<span
  className="ek-image"
/>
`;

exports[`<Image /> render 1`] = `
<img
  alt="test alt"
  className="ek-image"
  src="test.jpg"
/>
`;
